dev-util/xdelta/xdelta-3.1.0.recipe

```diff
@@ -0,0 +1,51 @@
+SUMMARY="A binary diff tool that uses the VCDIFF (RFC 3284) format and compression"
+DESCRIPTION="Xdelta is a tool and library for differential compression. Xdelta \
+release version 3 supports VCDIFF encoding and decoding."
+HOMEPAGE="http://xdelta.org/"
+COPYRIGHT="2001-2015 Josh MacDonald"
+LICENSE="Apache v2"
+REVISION="1"
+SOURCE_URI="https://github.com/jmacd/xdelta/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="7515cf5378fca287a57f4e2fee1094aabc79569cfe60d91e06021a8fd7bae29d"
+SOURCE_DIR="xdelta-$portVersion/xdelta3"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	xdelta$secondaryArchSuffix = $portVersion
+	cmd:xdelta3$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:liblzma$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:autoreconf
+	cmd:gcc$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	"
+
+BUILD()
+{
+	autoreconf --install
+	runConfigure ./configure
```

Comment:
I see two strategies for fixing a recipe that is being discussed in a PR.

The first strategy is to push incremental commits, generally one every time a change is requested. This helps the mentors to easily understand what changes were made at every step. And when everything looks OK, I would say that squashing all commits is a good practice.

The second strategy, which I use, is to squash my commits (and rebase them against master) everytime I have to fix something in a PR I have opened.
I also update the timestamp of my (squashed) commit to the current time just before I push it to my remote branch. This allows other contributors to easily notice that there has been a recent change and determine whether or not (s)he has already examined it.
That said, when I change only the commit message (or when I just want to rebase against master) then I don't update the timestamp.

I don't know if these practices are widely adopted. I came to the conclusion that this was a good strategy because it makes it easier for other contributors to pull remote branches (for testing purposes) without having to dive too far in the past. This is handy if, e.g. you want to push a tiny fix to the remote branch of a contributor who does not have time to update his PR.